### ISSUE TITLE
Warn if destination is unreachable

### ISSUE THREAD

**Issue by RenaKunisaki** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#321]
Recently I had an issue with some service/industry buildings failing to spawn vehicles (trucks in use goes from 0 to 1 and immediately back to 0), and after a lot of investigation, I discovered the cause: the only route to some buildings went through a road that had banned trucks. Removing the restriction solved the problem.

It would be a big help if TMPE reported (even just to `Debug.Log`) when pathfinding fails, where the source and destination are, so that we can more easily find such broken routes.

**Comment by CosignCosine** [comment]
I was under the impression that this operation would be extremely expensive when scaled to the entire vehicle simulation under a detour (although I may be wrong).

**Comment by aubergine10** [comment]
TMPE already shows which destinations cims are trying to get to on one of the info views (traffic, or routes, can't remember; Real Time does similar).

For showing the route though, we'd probably need a diagnostic tool. We can't just take the point at which pathfinder fails, as that probably won't highlight the problem (the patfinder will keep trying other routes beyond the main problem point).

Related issue: https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/issues/96

**Comment by RenaKunisaki** [comment]
Right, with this problem though the info views don't show any route because it can't find any. I'm talking about just a log message like "Failed to find route from Some Company (building 12345) to Some Shop (building 7511)", just to draw attention to the fact that a routing problem exists.

Personally I have the game's debug log displaying live on another screen, so even that would be enough for me. I assume you'd want some kind of in-game display of these messages too, though.

**Comment by aubergine10** [comment]
There's probably something along those lines if you build the mod from source in DEBUG setting.

**Comment by rogermartensson** [comment]
It would be really nice if this could be shown either in the "Failing building" or using a "search for a fix mod" with it's own window.

If possible it show the name and be clickable so you can "zoom into" the failing building. Atleast then one have a starting point to look for what is wrong.

A more advanced way would be to show where the most paths fail but that I guess would be pretty hard to show.

Anyway. There seems to be some problem with the pathfinding in the TM:PE-mod because the problem disappears when disabling the mod. 
I've tried to disable the all the settings (or changing if multivalued) and that doesn't help. If there was a button to disable the pathfinding while in-game that would be a good test.

**Comment by kvakvs** [comment]
Possible solution: Introduce a storage for N most recent problems per building id, or most recent in last minute, and when user opens a building, show a text label with the description of the last discovered problem or two. For example: `Delivery vehicle failed to find a path to destination`.

**Comment by rogermartensson** [comment]
I had two problems. One was due how I have set up my roads with TM:PE. It sure took a while to find it. :) The other was a Recycling center that got fixed when I turned off "highway lane rules". On this one I didn't find the "real reason of the problem".

The pathfinding isn't the problem here. It can't find the destination so it bails out. From my view as a user it seems that the when it fails it never tries anything else. For example. I had some resource storage that could've been used but when it was in "no path found"-mode it never got used. 

So if there was a way to show on the building that there is a problem finding a path to a building (with clickable name to "zoom" to building) that would be great.

I think someone with insight could take a look and see if the pathfinding could blacklist a building for delivery since there is no path to it. Then it wouldn't be stuck trying to spawn trucks.

If it's delivery that fails then the building needing material would be show as "not getting enough resources" so that would be a sign there is something wrong around that building. Same would be with other services. Other game play mechanics would show it's something wrong.

Then it would be possible to "debug" the city. ;-) This could be a bit hard because it's not that obvious that the problems is because of bad or wrong roads. So some sort of error at the resource producer would be nice.
